packages/calypso-analytics/src/index.js

```diff
@@ -0,0 +1,260 @@
+	// 9 * 4/3 = 12
+	// this is to avoid getting padding of a random byte string when it is base64 encoded
+	let randomBytes;
+
+	if ( window.crypto && window.crypto.getRandomValues ) {
+		randomBytes = new Uint8Array( randomBytesLength );
+		window.crypto.getRandomValues( randomBytes );
+	} else {
+		randomBytes = times( randomBytesLength, () => Math.floor( Math.random() * 256 ) );
+	}
+
+	return window.btoa( String.fromCharCode.apply( String, randomBytes ) );
+}
+
+function checkForBlockedTracks() {
+	// Proceed only after the tracks script load finished and failed.
+	// Calling this function from `initialize` ensures current user is set.
+	// This detects stats blocking, and identifies by `getCurrentUser()`, URL, or cookie.
+	_loadTracksResult.catch( () => {
+		let _ut, _ui;
+		const currentUser = getCurrentUser();
+
+		if ( currentUser && currentUser.ID ) {
+			_ut = 'wpcom:user_id';
+			_ui = currentUser.ID;
+		} else {
+			_ut = getUrlParameter( '_ut' ) || 'anon';
+			_ui = getUrlParameter( '_ui' );
+
+			if ( ! _ui ) {
+				const cookies = cookie.parse( document.cookie );
+				if ( cookies.tk_ai ) {
+					_ui = cookies.tk_ai;
+				} else {
+					const randomIdLength = 18; // 18 * 4/3 = 24 (base64 encoded chars).
+					_ui = createRandomId( randomIdLength );
+					document.cookie = cookie.serialize( 'tk_ai', _ui );
+				}
+			}
+		}
+
+		blockedTracksDebug( 'Loading /nostats.js', { _ut, _ui } );
+		return loadScript(
+			'/nostats.js?_ut=' + encodeURIComponent( _ut ) + '&_ui=' + encodeURIComponent( _ui )
+		);
+	} );
+}
+
+export const analyticsEvents = new EventEmitter();
+
+/**
+ * Returns the anoymous id stored in the `tk_ai` cookie
+ *
+ * @returns {string} - The Tracks anonymous user id
+ */
+export function getTracksAnonymousUserId() {
+	const cookies = cookie.parse( document.cookie );
+
+	return cookies.tk_ai;
+}
+
+export function initializeAnalytics( currentUser, superProps ) {
+	// Update super props.
+	if ( 'function' === typeof superProps ) {
+		initializeDebug( 'superProps', superProps );
+		_superProps = superProps;
+	}
+
+	// Identify current user.
+	if ( 'object' === typeof currentUser ) {
+		initializeDebug( 'identifyUser', currentUser );
+		identifyUser( currentUser );
+	}
+
+	// Tracks blocked?
+	initializeDebug( 'checkForBlockedTracks' );
+	checkForBlockedTracks();
+}
+
+export function identifyUser( userData ) {
+	// Ensure object.
+	if ( 'object' !== typeof userData ) {
+		identifyUserDebug( 'Invalid userData.', userData );
+		return; // Not possible.
+	}
+
+	// Set current user.
+	const currentUser = setCurrentUser( userData );
+	if ( ! currentUser ) {
+		identifyUserDebug( 'Insufficient userData.', userData );
+		return; // Not possible.
+	}
+
+	// Tracks user identification.
+	identifyUserDebug( 'Tracks identifyUser.', currentUser );
+	window._tkq.push( [ 'identifyUser', currentUser.ID, currentUser.username ] );
+}
+
+export function recordTracksEvent( eventName, eventProperties ) {
```

Comment:
This `src/index.js` file feels very disorganized -- a random heap of functions and constants rather than a clean package API.

Can we move all Tracks code to a dedicated module, e.g., `src/tracks.js` and use `src/index.js` only for re-exporting things?

Reply:
I made that change and it looks much better :)